### PR TITLE
Added rechunk method to DaskSpectralCube

### PR DESCRIPTION
**Pull request by astrofrog**
This also adds the chunk size to the DaskSpectralCube repr:

```
    DaskSpectralCube with shape=(7, 4, 3) and unit=Jy / beam and chunk size (7, 4, 3):
     n_x:      3  type_x: RA---ARC  unit_x: deg    range:    52.231466 deg:   52.231544 deg
     n_y:      4  type_y: DEC--ARC  unit_y: deg    range:    31.243639 deg:   31.243739 deg
     n_s:      7  type_s: VRAD      unit_s: m / s  range:    14322.821 m / s:   14944.909 m / s
```

### CODE REVIEW

File: docs/dask.rst [hunk 1]
```diff
@@ -92,6 +92,37 @@ installed.
 This can also be beneficial if you are using multiprocessing or multithreading to carry out calculations,
 because zarr works nicely with disk access from different threads and processes.
 
+Rechunking data
+---------------
+
+In some cases, the way the data is chunked on disk may be inefficient (for example large CASA
+datasets may be chunked into tens of thousands of blocks, which may make dask operations slow due to
```
Comment (keflavich) [inline docs/dask.rst:99]:
```suggestion
datasets may be chunked into tens of thousands of blocks), which may make dask operations slow due to
```

File: spectral_cube/dask_spectral_cube.py [hunk 2]
```diff
@@ -310,6 +310,45 @@ def _get_filled_data(self, view=(), fill=np.nan, check_endian=None, use_memmap=N
...
+                **kwargs):
+        """
+        Rechunk the underlying dask array and return a new cube.
+
+        Parameters
+        ----------
+        chunks:  int, tuple, dict or str, optional
+            The new block dimensions to create. -1 indicates the full size of the
+            corresponding dimension. Default is "auto" which automatically
+            determines chunk sizes.
```
Comment (e-koch) [inline spectral_cube/dask_spectral_cube.py:330]: We should mention here and/or in the docs that you can mix some input types, too. e.g., `chunks=(-1, 'auto', 'auto')` is ideal for moment maps.

File: spectral_cube/dask_spectral_cube.py [hunk 3]
```diff
@@ -310,6 +310,45 @@ def _get_filled_data(self, view=(), fill=np.nan, check_endian=None, use_memmap=N
...
+        default_repr = super().__repr__()
+        lines = default_repr.splitlines()
+        lines[0] = lines[0][:-1] + ' and chunk size {0}:'.format(self._data.chunksize)
+        return '\n'.join(lines)
+
+    @add_save_to_tmp_dir_option
+    def rechunk(self, chunks='auto', threshold=None, block_size_limit=None,
+                **kwargs):
+        """
+        Rechunk the underlying dask array and return a new cube.
```
Comment (e-koch) [inline spectral_cube/dask_spectral_cube.py:323]: A link to the dask array chunks docs here would be useful for more explanation of the kwargs. `threshold` seems hard to understand without some understanding of what dask is doing.

https://docs.dask.org/en/latest/array-chunks.html

File: docs/dask.rst [hunk 4]
```diff
@@ -92,6 +92,37 @@ installed.
...
+
+or by specifying it explicitly::
+
+    >>> cube_new = cube.rechunk(chunks=(2, 2, 2), save_to_tmp_dir=True)  # doctest: +IGNORE_OUTPUT
+    [########################################] | 100% Completed |  0.1s
+    >>> cube_new
+    DaskSpectralCube with shape=(7, 4, 3) and unit=Jy / beam and chunk size (2, 2, 2):
+     n_x:      3  type_x: RA---ARC  unit_x: deg    range:    52.231466 deg:   52.231544 deg
+     n_y:      4  type_y: DEC--ARC  unit_y: deg    range:    31.243639 deg:   31.243739 deg
+     n_s:      7  type_s: VRAD      unit_s: m / s  range:    14322.821 m / s:   14944.909 m / s
```
Comment (e-koch) [inline docs/dask.rst:120]: Either now or later, we should show rechunking for full spectra (`(-1, 'auto', 'auto')`) and the image plane (`('auto', -1, -1)`). These seem like some of the most common use cases for optimizing with rechunking.

Comment (astrofrog) [reply docs/dask.rst:120]: Done in narrative docs

File: spectral_cube/dask_spectral_cube.py (hunk 3 above)
Comment (astrofrog) [reply spectral_cube/dask_spectral_cube.py:323]: Done

File: spectral_cube/dask_spectral_cube.py (hunk 2 above)
Comment (astrofrog) [reply spectral_cube/dask_spectral_cube.py:330]: Done

**Comment by astrofrog** [comment]
@keflavich @e-koch - should be good to go!

**Pull request merged**